### PR TITLE
📝 `Developing` Guide page for the `docs`

### LINKED ISSUE #340: 📝 `Developing` Guide page for the `docs`
### Description

Provide an in-depth `developing guide` page in the `docs` located under the codebase (`websurfx` directory) explaining how to work on the project, what is the general workflow to follow and what are the different ways to work on the project. Also, providing steps and examples if necessary.

Example of a good `developing guide`:
- https://github.com/Lissy93/dashy/blob/master/docs/developing.md

### Screenshots

_No response_

### Do you want to work on this issue?

None

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by neon-mmd**
## What does this PR do?

This PR provides a new in-depth development guide page for the contributors.

## Why is this change important?

This change is essential as it guides developers (beginner or advanced) into the project like what are the different ways to work on the project, how to set up the development environment, project structure, project workflow, etc.

## Author's checklist

- [x] Provide an in-depth development guide for contributors.
- [x] Bump the app version to `v1.2.4` 

## Related issues

Closes #340

### CODE REVIEW

File: docs/developing.md [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+
+<details>
+  <summary><b>Table of Contents</b></summary>
+  <p>
+
+- [Setting up the Development Environment](#setting-up-the-development-environment)
+  - [Local Development](#local-development-)
+  - [Gitpod](#gitpod-)
+  - [NixOS Dev Shell using Nix Flake](#nixos-dev-shell-using-nix-flake-)
+  - [Local Development with Docker Compose](#local-development-with-docker-compose-)
```
Comment (alamin655) [inline docs/developing.md:13]:
```suggestion
  - [Local Development](#local-development)
  - [Gitpod](#gitpod)
  - [NixOS Dev Shell using Nix Flake](#nixos-dev-shell-using-nix-flake)
  - [Local Development with Docker Compose](#local-development-with-docker-compose)
```

File: docs/developing.md [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+  - [Project Commands](#project-commands)
+  - [Environmental Variables](#environmental-variables)
```
Comment (alamin655) [inline docs/developing.md:15]:
```suggestion
  - [Environment Variables](#environment-variables)
```

File: docs/developing.md [hunk 3]
```diff
@@ -0,0 +1,629 @@
...
+#### Development
+
+- `cargo watch -q -x "run" -w .`: Starts the development server with hot reloading.
+- `cargo fmt -- --check`: Checks the code for proper formatting.
+- `cargo clippy`: Lints code to ensure it follows a consistent, neat style.
+- `cargo test`: Runs unit tests, integrations tests and doc tests.
+
+### Environment Variables
+
+All environmental variables are optional. Currently there are not many environmental variables used, as most of the user preferences are stored under the `websurfx` folder (located under the codebase (`websurfx` directory)) in the `config.lua` file.
```
Comment (alamin655) [inline docs/developing.md:385]:
```suggestion
Environment variables are optional. Currently there are not many environment variables used, as most of the user preferences are stored under the `websurfx` folder (located under the codebase (`websurfx` directory)) in the `config.lua` file.
```

File: docs/developing.md [hunk 4]
```diff
@@ -0,0 +1,629 @@
...
+  - [Project Commands](#project-commands)
+  - [Environmental Variables](#environmental-variables)
+- [Git Strategy](#git-strategy)
+  - [Flow](#git-flow)
+  - [Branches](#git-branch-naming)
+  - [Commit emojis](#commit-emojis)
+  - [PR Guidelines](#pr-guidelines)
+- [Resources for Beginners](#resources-for-beginners)
+- [App Info](#app-info)
+- [Code Style Guide](#style-guide)
```
Comment (alamin655) [inline docs/developing.md:23]:
```suggestion
- [Code Style Guide](#style-guides)
```

**Review by alamin655 — COMMENTED** [review]
My one suggestion will be to reduce the commit emojis to 4-5 that we are currently using.

**Comment by neon-mmd** [comment]
Thanks ❤️ for taking the time reviewing this PR but I think I would suggest having all the suggestions as a good first issue becauss I think it woulf be better. What do you say ? 🙂

**Comment by alamin655** [comment]
> Thanks ❤️ for taking the time reviewing this PR but I think I would suggest having all the suggestions as a good first issue becauss I think it woulf be better. What do you say ? 🙂

Okay, then that would be better.

**Review by alamin655 — APPROVED** [review]
Approved! Let's merge this in. 👍

**Comment by neon-mmd** [comment]
> My one suggestion will be to reduce the commit emojis to 4-5 that we are currently using.

No actually,  I think having more emojis would be better because it will make distinguishing between different changes more easier 🙂 . What do you think?

**Comment by alamin655** [comment]
> > My one suggestion will be to reduce the commit emojis to 4-5 that we are currently using.
> 
> No actually, I think having more emojis would be better because it will make distinguishing between different changes more easier 🙂 . What do you think?

Just not my thing. I think minimal is better, but I will leave it to you to decide.

**Comment by neon-mmd** [comment]
> > > My one suggestion will be to reduce the commit emojis to 4-5 that we are currently using.
> > 
> > 
> > No actually, I think having more emojis would be better because it will make distinguishing between different changes more easier 🙂 . What do you think?
> 
> Just not my thing. I think minimal is better, but I will leave it to you to decide.

I think having more would be better so lets go with that 🙂 .

**Pull request merged**